Add zod as dependency for example todolist plugin to specify version

clairelcasey opened the pull request:
## Hey, I just made a Pull Request!

This specifies the zod version for the example permissions todolist plugin. This is needed because there were TS errors with the newest version of zod which gets installed from the CLI.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

freben commented:
Hi! Note that the CI checks will say that you need to run `yarn install` and commit the `yarn.lock` changes too.

But regarding the fix: The article isn't just for the example TODO backend, but can be used generally, right? So removing `zod` from the installation instruction will make things worse for those who follow along in behalf of their own plugin. I'd suggest to either

- do nothing (if it's likely that `zod` will have a fix issued soon) and tell people to change their version in package.json in the mean time, or
- update the `yarn workspace` command to say `zod@~3.18.0` at the end explicitly, and remove that bit when it's no longer necessary

The latter is my favorite, but I'll leave it up to you of course.

clairelcasey commented:
Thanks @freben!! Great call-out, just updated with your second option which is my preference too :) also added a reminder for myself to check in on zod updates in a month!

freben reviewed (approved):
Nice.
